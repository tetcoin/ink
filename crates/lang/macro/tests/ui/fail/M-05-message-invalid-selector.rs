use pro_lang as pro;

#[pro::contract]
mod message_invalid_selector {
    #[pro(storage)]
    pub struct MessageInvalidSelector {}

    impl MessageInvalidSelector {
        #[pro(constructor)]
        pub fn constructor() -> Self {
            Self {}
        }

        #[pro(message, selector = "0x00")]
        pub fn invalid_selector(&self) {}
    }
}

fn main() {}
