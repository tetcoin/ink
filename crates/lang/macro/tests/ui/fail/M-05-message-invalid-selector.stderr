error: invalid selector - a selector must consist of four bytes in hex (e.g. `selector = "0xCAFEBABE"`)
  --> $DIR/M-05-message-invalid-selector.rs:14:24
   |
14 |         #[pro(message, selector = "0x00")]
   |                        ^^^^^^^^^^^^^^^^^
